src/main/java/com/amazonaws/kinesisvideo/internal/producer/jni/NativeKinesisVideoProducerJni.java

```diff
@@ -506,6 +506,25 @@ public void putFragmentMetadata(final long streamHandle, final @Nonnull String m
         putKinesisVideoFragmentMetadata(mClientHandle, streamHandle, metadataName, metadataValue, persistent);
     }
 
+    /**
+     * Put an event metadata.
+     *
+     * @param streamHandle  The handle of the stream.
+     * @param event  The type of event(s), a value from the StreamEventType enum.
+ *                   If you want to submit multiple events in one call it is suggested to use bit-wise
+ *                   OR combination from StreamEventType enum.
+     * @param streamEventMetadata  Optional metadata. This metadata will be applied
+ *                                 to all events included in THIS function call.
+     * @throws ProducerException
+     */
+    public void putEventMetadata(final long streamHandle, final int event, final @Nullable StreamEventMetadata streamEventMetadata)
```

Comment:
PIC does not enforce that this parameter be of StreamEventType and neither should we here because:
1. We allow for combining StreamEventType values using bitwise AND which would give us a non-StreamEventType value to be passed here.
2. Code parity.